src/account.c

```diff
@@ -1711,47 +1711,61 @@ const char *account_extra(const struct account *acc)
 /**
  * Auto complete a SIP uri, add scheme and domain if missing
  *
- * @param acc User-Agent account
- * @param buf Target buffer to print SIP uri
- * @param uri Input SIP uri
+ * @param acc  User-Agent account
+ * @param strp Pointer to destination string; allocated and set
+ * @param uri  Input SIP uri pointer-length string
  *
  * @return 0 if success, otherwise errorcode
  */
-int account_uri_complete(const struct account *acc, struct mbuf *buf,
-			 const char *uri)
+int account_uri_complete(const struct account *acc, char **strp,
+			 const struct pl *uri)
 {
 	struct sa sa_addr;
-	size_t len;
 	bool uri_is_ip;
 	char *uridup;
 	char *host;
 	char *c;
+	bool append = true;
+	struct mbuf *mb;
+	struct pl trimmed;
+	bool complete = false;
 	int err = 0;
 
-	if (!buf || !uri)
+	if (!strp || !pl_isset(uri))
 		return EINVAL;
 
 	/* Skip initial whitespace */
-	while (isspace(*uri))
-		++uri;
+	trimmed = *uri;
+	err = pl_ltrim(&trimmed);
+	if (err)
+		return err;
 
-	len = str_len(uri);
+	mb = mbuf_alloc(64);
+	if (!mb)
+		return ENOMEM;
 
 	/* Append sip: scheme if missing */
-	if (0 != re_regex(uri, len, "sip:"))
-		err |= mbuf_printf(buf, "sip:");
-
-	err |= mbuf_write_str(buf, uri);
+	if (re_regex(trimmed.p, trimmed.l, "sip:"))
+		err |= mbuf_printf(mb, "sip:");
+	else
+		complete = true;
 
+	err |= mbuf_write_pl(mb, &trimmed);
 	if (!acc || err)
-		return err;
+		goto out;
 
-	/* Append domain if missing and uri is not IP address */
+	if (complete)
+		goto out;
 
+	conf_get_bool(conf_cur(), "append_domain", &append);
```

Comment:
this logic should be outside libbaresip core

It should be a specific config item for "menu" module.

Reply:
Then this setting makes no sense any longer.
We will avoid the automatic host part append by appending the host part in the application if it makes sense for us, or prepend the scheme "sip:".

This PR turns into a cleanup PR for `menu`. Following PRs could also reduce the code for `gtk`.